hugo/content/tutorials/langium_and_monaco.md

```diff
@@ -169,233 +214,368 @@ h1 {
 
-```js
-const editorConfig = {
-    languageId,
-    code,
-    useDiffEditor: false,
-    automaticLayout: true,
-    theme: 'vs-dark'
-};
+```ts
+/**
+ * Creates & returns a fresh worker using the MiniLogo language server
+ */
+function getWorker() {
+    const workerURL = new URL('minilogo-server-worker.js', window.location.href);
+    return new Worker(workerURL.href, {
+        type: 'module',
+        name: 'MiniLogoLS'
+    });
+}
 ```
 
-Since we're planning to use a language server with Monaco, we'll need to setup a language client config too. To do this we'll also need to generate a worker using our language server worker file, but that's fairly straightforward to setup here.
+By creating the worker in advance, we give ourselves the ability to directly interact with the worker/LS independent of the wrapper itself, and to even pre-configure it before use. This can be hugely beneficial, especially if we expect to customize our LS on the fly.
 
-```js
-// configure the worker
-const workerURL = new URL('./minilogo-server-worker.js', import.meta.url);
-const lsWorker = new Worker(workerURL.href, {
-    type: 'classic',
-    name: 'minilogo-language-server-worker'
-});
+Lastly, let's setup the user config, which will be used to startup the wrapper.
 
-// setup the language client config with the worker
-const languageClientConfig = {
-    enabled: true,
-    useWebSocket: false,
-    // can pass configuration data, or a pre-configured worker as well
-    // the later works better for us in this case
-    workerConfigOptions: lsWorker
-};
-```
+```ts
+type WorkerUrl = string;
 
-We can also pass more explicit config options to `workerConfigOptions` if we don't want to create the worker ourselves. For example:
+/**
+ * Generalized configuration used with 'getMonacoEditorReactConfig' to generate a working configuration for monaco-editor-react
+ */
+interface ClassicConfig {
+    code: string,
+    htmlElement: HTMLElement,
+    languageId: string,
+    worker: WorkerUrl | Worker,
+    monarchGrammar: any;
+}
 
-```js
-{
-    url: new URL('./minilogo-server-worker.js', import.meta.url),
-    type: 'classic',
-    name: 'minilogo-language-server-worker'
+/**
+ * Generates a UserConfig for a given Langium example, which is then passed to the monaco-editor-react component
+ * 
+ * @param config A VSCode API or classic editor config to generate a UserConfig from
+ * @returns A completed UserConfig
+ */
+function createUserConfig(config: ClassicConfig): UserConfig {
+    // setup urls for config & grammar
+    const id = config.languageId;
+
+    // generate langium config
+    return {
+        htmlElement: config.htmlElement,
+        wrapperConfig: {
+            editorAppConfig: {
+                $type: 'classic',
+                languageId: id,
+                useDiffEditor: false,
+                automaticLayout: true,
```

Comment:
Btw, this is optional and `automaticLayout` is used if not specified.

Reply:
I've noticed this in some of the new examples, but I need to apply it here, thanks for pointing that out.